utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -499,5 +503,183 @@ void PessimisticTransactionDB::UnregisterTransaction(Transaction* txn) {
   transactions_.erase(it);
 }
 
+// Returns true if commit_seq <= snapshot_seq
+bool WritePreparedTxnDB::IsInSnapshot(uint64_t prep_seq,
```

Comment:
Good point. The current implementation is certainly inefficient. The idea is to have a correct patch landed to unblock all the next parallel tasks that depend on it. Optimization will come after.